v2/dtos/requests/deviceprofile_test.go

```diff
@@ -0,0 +1,243 @@
+		{"invalid AddDeviceProfileRequest, no command name", noCommandName, true},
+		{"invalid AddDeviceProfileRequest, no command Get", noCommandGet, true},
+		{"invalid AddDeviceProfileRequest, no command Put", noCommandPut, true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.DeviceProfile.Validate()
+			assert.Equal(t, tt.expectError, err != nil, "Unexpected addDeviceProfileRequest validation result.", err)
+		})
+	}
+}
+
+func TestAddDeviceProfile_UnmarshalJSON(t *testing.T) {
+	valid := testAddDeviceProfileReq
+	resultTestBytes, _ := json.Marshal(testAddDeviceProfileReq)
+	type args struct {
+		data []byte
+	}
+	tests := []struct {
+		name     string
+		expected AddDeviceProfileRequest
+		args     args
+		wantErr  bool
+	}{
+		{"unmarshal AddDeviceProfileRequest with success", valid, args{resultTestBytes}, false},
+		{"unmarshal invalid AddDeviceProfileRequest, empty data", AddDeviceProfileRequest{}, args{[]byte{}}, true},
+		{"unmarshal invalid AddDeviceProfileRequest, string data", AddDeviceProfileRequest{}, args{[]byte("Invalid AddDeviceProfileRequest")}, true},
+	}
+	fmt.Println(string(resultTestBytes))
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var dp AddDeviceProfileRequest
+			err := dp.UnmarshalJSON(tt.args.data)
+			if tt.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.expected, dp, "Unmarshal did not result in expected AddDeviceProfileRequest.")
+			}
+		})
+	}
+}
+
+func TestAddDeviceProfile_UnmarshalYAML(t *testing.T) {
+	valid := testAddDeviceProfileReq
+	resultTestBytes, _ := yaml.Marshal(testAddDeviceProfileReq)
+	type args struct {
+		data []byte
+	}
+	tests := []struct {
+		name     string
+		expected AddDeviceProfileRequest
+		args     args
+		wantErr  bool
+	}{
+		{"unmarshal AddDeviceProfileRequest with success", valid, args{resultTestBytes}, false},
+		{"unmarshal invalid AddDeviceProfileRequest, empty data", AddDeviceProfileRequest{}, args{[]byte{}}, true},
+		{"unmarshal invalid AddDeviceProfileRequest, string data", AddDeviceProfileRequest{}, args{[]byte("Invalid AddDeviceProfileRequest")}, true},
+	}
+	fmt.Println(string(resultTestBytes))
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var dp AddDeviceProfileRequest
+			err := dp.UnmarshalYAML(tt.args.data)
+			if tt.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, tt.expected, dp, "Unmarshal did not result in expected AddDeviceProfileRequest.")
+			}
+		})
+	}
+}
+
+func Test_AddDeviceProfileReqToDeviceProfileModels(t *testing.T) {
+	valid := []AddDeviceProfileRequest{testAddDeviceProfileReq}
+	expectedDeviceProfileModels := []models.DeviceProfile{expectedDeviceProfile}
+	tests := []struct {
+		name              string
+		addDeviceProfiles []AddDeviceProfileRequest
+	}{
+		{"valid AddDeviceProfileRequest", valid},
```

Comment:
Do you plan to add invalid test case? If, not change to non-table driven to reduce complexity.

Reply:
Changed to non-table driven testing.